### ISSUE TITLE
SetShapeFn gets nullptr for custom operator leasing to crash

### ISSUE THREAD

**Issue by JanuszL** [tensorflow/tensorflow#29643]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NA
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.14.0rc1
- Python version: 2.7
- Bazel version (if compiling from source): NA
- GCC/Compiler version (if compiling from source): gcc 7.4.0
- CUDA/cuDNN version: NR
- GPU model and memory: NR

**Describe the current behavior**
`SetShapeFn` doesn't work when called for `REGISTER_OP` for the custom operators. Whenever it is called by TF is gets `nullptr` instead valid value of the `InferenceContext` argument.

**Describe the expected behavior**
`SetShapeFn` should get meaningful ptr.

**Code to reproduce the issue**
It doesn't work with https://www.tensorflow.org/guide/extend/op. [ Repro attached](https://github.com/tensorflow/tensorflow/files/3276625/tf_crash_repro.zip), just call `run.sh`.

**Other info / logs**
NA

**Comment by tfboyd** [comment]
@lamberta Can you find someone to verify this and find an owner?

**Comment by lamberta** [comment]
@yifeif has been working on the https://github.com/tensorflow/custom-op repo so maybe can take a look.

Talking with @MarkDaoust and maybe https://www.tensorflow.org/guide/extend/op should just move over to that repo.

**Comment by JanuszL** [comment]
It doesn't work with https://github.com/tensorflow/custom-op after calling `make zero_out_test`. But it is just a repro, the issue was reported because our custom operator doesn't work as well.

**Comment by samikama** [comment]
@JanuszL , I guess the issue in the example is explicit addition of CXX abi to the compiler flags. You should not add this since TF already gives that flag and may conflict with what you specify. When I removed the explicit -D_GLIBCXX_USE_CXX11_ABI=0 from your example, it worked fine for me. Could you please check whether in your custom op you are not overriding it?

**Comment by JanuszL** [comment]
@samikama - removing `-D_GLIBCXX_USE_CXX11_ABI=0` doesn't work either. Issuing:
```
python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_compile_flags()))'
```
gets me
```
-I/usr/local/lib/python3.6/dist-packages/tensorflow/include -D_GLIBCXX_USE_CXX11_ABI=0
```
And  https://github.com/tensorflow/custom-op  doesn't work as well, while there is no ` -D_GLIBCXX_USE_CXX11_ABI` flag passed explicitly.

**Comment by samikama** [comment]
@JanuszL you are right. Unless you use gcc 4.8 it bombs. @tfboyd @gunan This looks like an ABI issue. Was there a change in the way official pip package built for 1.14? I remember centos6 discussion but not sure if it made into 1.14. If it did, there seems to be a serious issue here.

**Comment by gunan** [comment]
@yifeif for custom op instructions.
@samikama our pip packages are still built with gcc 4.8.
Unfortuanately, this is out of our control. To make anything that uses C++ APIs work, all custom ops need to be built with the same compiler (and the version) we use to build the pip packages. custom-op repository has a docker container with our toolchains deployed. Anything not built with that may break due to compilers generating ABIs differently.

@yifeif should be able to help more on this.

**Comment by Tombana** [comment]
> custom-op repository has a docker container with our toolchains deployed. Anything not built with that may break due to compilers generating ABIs differently.

The custom-op repository references two docker images. The manylinux2010 image `tensorflow/tensorflow:custom-op-ubuntu16` has gcc 5.4 and indeed gives `nullptr` for custom op shape functions. The image `tensorflow/tensorflow:custom-op-ubuntu14` has gcc 4.8 and works fine with custom ops.

**Comment by zh794390558** [comment]
when to unify toolchains of  the pip build docker and released docker images ?

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this issue, as you are using an older version of tensorflow(1.x) which is officially considered as end of life. We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, Please open a new issue for any help you need against 2.x, and we will get you the right help.
